### PR TITLE
MNT - Removes Python 3.8 support

### PR DESCRIPTION
**Pull request by lazarust**

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -10,18 +10,15 @@ concurrency:
...
-      fail-fast: false  # need to see which ones fail
+      fail-fast: false # need to see which ones fail
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        python: ["3.8", "3.9", "3.10", "3.11"]
+        python: ["3.9", "3.10", "3.11"]
         # this is to make the CI run on different sklearn versions
         include:
-          - python: "3.8"
-            sklearn_version: "1.0"
```
Comment (adrinjalali) [inline .github/workflows/build-test.yml:24]: in the same vein as others, could you please add py312 and sklearn 1.4?

Comment (lazarust) [reply .github/workflows/build-test.yml:24]: Done! I also had to pin the macOS version in the action from @BenjaminBossan's comment: https://github.com/skops-dev/skops/pull/394#issuecomment-2082111730

File: setup.py [hunk 2]
```diff
@@ -67,13 +67,12 @@ def setup_package():
             "Operating System :: Unix",
             "Operating System :: MacOS",
             "Programming Language :: Python :: 3",
-            "Programming Language :: Python :: 3.8",
             "Programming Language :: Python :: 3.9",
             "Programming Language :: Python :: 3.10",
             "Programming Language :: Python :: 3.11",
```
Comment (adrinjalali) [inline setup.py:72]:
```suggestion
            "Programming Language :: Python :: 3.11",
            "Programming Language :: Python :: 3.12",
```

**Review by adrinjalali — COMMENTED** [review]
I'm confused as why the diff looks so large, but other than the nit looks good to me.

**Review by adrinjalali — APPROVED** [review]

**Pull request merged**